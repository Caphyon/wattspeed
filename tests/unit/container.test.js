import { createLocalVue, shallowMount } from "@vue/test-utils";
import container from "../../src/components/panels/tech/container";

const localVue = createLocalVue();

describe("Testing Container component", () => {
  it("Container with data", () => {
    const wrapper = shallowMount(container, {
      localVue,
      computed: {
        title() {
          return "SECTION";
        },
        desc() {
          return "Descriere Test"
        },
      },
    });

    // controll if the component is instance
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();

    // check if the divs are presents
    expect(wrapper.find(".mainContent").exists()).toBe(true);
    expect(wrapper.find(".section__header").exists()).toBe(true);
    expect(wrapper.findAll(".sections__container--title").length).toBe(1);
    expect(wrapper.find("a.close[aria-label=\"Close button\"]").exists()).toBe(true);
    expect(wrapper.find(".section__header p.text--center").exists()).toBe(true);

    const spy = spyOn(wrapper.vm, 'close');
    wrapper.find('a.close').trigger('click');
    expect(spy).toBeCalled();
    wrapper.destroy();
  });

});
