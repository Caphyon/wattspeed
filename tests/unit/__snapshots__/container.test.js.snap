// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Container component Container with data 1`] = `
<div
  class="mainContent"
>
  <div
    class="section__header"
  >
    <div
      class="sections__container--title"
    >
      <svg
        class="section--icon"
        height="20"
        width="20"
      >
        <use
          xlink:href="assets/icons/symbols.svg#undefined"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        />
      </svg>
       
      <h4
        class="mt0 mb0 text--strong"
      >
        SECTION
      </h4>
    </div>
     
    <a
      aria-label="Close button"
      class="close"
      href="#"
      title="Close"
    >
      Ã—
    </a>
      
    <p
      class="sections__container--description text--center text--small px2 py1"
    >
      Descriere Test
    </p>
  </div>
   
  <!---->
   
  <!---->
</div>
`;
